### ISSUE TITLE
fulltest failed

### ISSUE THREAD

**Issue by dannypsnl** [llir/llvm#196]
@mewmew could you take a look? Changes destroyed test
```
ir/enum/tools.go:6:2: import "golang.org/x/tools/cmd/stringer" is a program, not an importable package
ir/types/tools.go:6:2: import "golang.org/x/tools/cmd/stringer" is a program, not an importable package
```

**Comment by mewmew** [comment]
Asked upstream for help on what the current best practice is for handling tools dependencies: https://github.com/golang/go/issues/25922#issuecomment-883181614

**Comment by mewmew** [comment]
Upstream answer from @jayconrod.

> @mewmew In this case, don't run go vet with -tags tools. The tools.go file is not meant to be part of a buildable package, so vet won't provide useful feedback on it. Without -tags tools, everything except go mod tidy will ignore tools.go.

@dannypsnl, we should update our CI test scripts to not include the `tools` tag for the `go vet` command. Could you look into how to do this?

Cheers,
Robin

**Comment by dannypsnl** [comment]
But we didn't run `go vet` command

**Comment by mewmew** [comment]
@dannypsnl, can you re-produce this issue locally?

When i run `go install -v ./...` I get no error locally (on the `llvm12` branch).

```bash
$ go install -v ./... ; echo $?
0
```

**Comment by mewmew** [comment]
Ahh, I think I know what the issue is. We updated the Go build tags to the format supported by Go 1.17 (i.e. using `//go:build tags` instead of `//+build tags`). Older versions of Go does not recognize the new Go build tag format.

So for the time being, we should use both it seems. Then when Go version 1.18 (or 1.19) is released, we can remove the use of `//+build` and only keep `//go:build`.

**Comment by mewmew** [comment]
Note, the build passes (on rev 11e88e3b2edd2aa8198e7b1434f2e571ce4414ac) using Go version 1.17.

```
u@x1 ~/l/p/g/l/llvm (llvm12)> go test ./...
?   	github.com/llir/llvm	[no test files]
ok  	github.com/llir/llvm/asm	20.987s
?   	github.com/llir/llvm/asm/enum	[no test files]
?   	github.com/llir/llvm/cmd/l-tm	[no test files]
ok  	github.com/llir/llvm/internal/enc	0.002s
?   	github.com/llir/llvm/internal/gep	[no test files]
ok  	github.com/llir/llvm/internal/natsort	0.020s
?   	github.com/llir/llvm/internal/osutil	[no test files]
ok  	github.com/llir/llvm/ir	0.173s
ok  	github.com/llir/llvm/ir/constant	0.927s
?   	github.com/llir/llvm/ir/enum	[no test files]
ok  	github.com/llir/llvm/ir/metadata	0.176s
ok  	github.com/llir/llvm/ir/types	0.005s
?   	github.com/llir/llvm/ir/value	[no test files]
```

**Comment by mewmew** [comment]
Fixed in ee6f05b8ce5a75cd1537738c7db1c3a7d5ad4b3b.
